### PR TITLE
bsunanda:Run2-alca46 Correct AlCaReco for NZS and bug fix in Analyzer

### PR DESCRIPTION
**Pull request by bsunanda**
Modify the reconstruction sequence for pedestal as well as signals needed for studies of NZS data. Also do bug fixing to the code which carries out phi symmetry analysis

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mmusich** [comment]
@bsunanda  would you please add some description of this PR?

**Comment by bsunanda** [comment]
@mmusich Done that

**Comment by mmusich** [comment]
@bsunanda thanks.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@mmusich @davidlange6 Could you approve this?

**Comment by mmusich** [comment]
@bsunanda starting to review it now.

**Comment by jozzez1** [comment]
I cross-checked the PR using [this script](https://github.com/jozzez1/AlCaTools/blob/master/scripts/CheckPR/CheckPullRequest.py)
```
$ xrdcp root://xrootd.unl.edu//store/data/Run2016B/HcalNZS/RAW/v2/000/273/158/00000/10E13AC8-7418-E611-818E-02163E01425B.root .
$ python CheckPullRequest.py CMSSW_8_1_0_pre6 14595
```
then in both versions, vanilla and changes I ran
```
$ sh command.sh
```

the contents of command.sh are:
```
$ cat command.sh
cmsDriver.py step2  --datatier FEVTDEBUGHLT --conditions auto:run2_hlt_relval -s L1REPACK:Full,HLT:@relval2016 --process reHLT --data  --era Run2_2016 --eventcontent FEVTDEBUGHLT -n 1000 --filein file:../../../../10E13AC8-7418-E611-818E-02163E01425B.root --fileout file:step2.root

cmsDriver.py step3  --conditions auto:run2_data_relval -s RAW2DIGI,L1Reco,RECO,ALCA:HcalCalMinBias --runUnscheduled  --process reRECO --data  --era Run2_2016 --eventcontent RECO --hltProcess reHLT --scenario pp --datatier RECO -n 1000 --customise Configuration/DataProcessing/RecoTLR.customisePostEra_Run2_2016 --filein file:step2.root --fileout file:step3.root 
```

Files ALCARECOHcalCalMinBias.root are equivallent with respect to `edmDumpEventContent` and `edmEventSize -v -a`

**Comment by mmusich** [comment]
thanks @jozzez1 for the detailed checks

**Comment by mmusich** [comment]
+1

File: Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalMinBias_cff.py [hunk 1]
```diff
@@ -7,6 +7,10 @@
 import EventFilter.HcalRawToDigi.HcalRawToDigi_cfi
 hcalDigiAlCaMB = EventFilter.HcalRawToDigi.HcalRawToDigi_cfi.hcalDigis.clone()
 import RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_hbhe_cfi
+hbherecoMBspecial = RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_hbhe_cfi.hbheprereco.clone()
```
Comment (davidlange6) [inline Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalMinBias_cff.py]: @bsunanda - would it be easy to change "special" to something more physically meaningful? (for future code readability)

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@mmusich @davidlange6 Changed name of ..special.. to ...NZS...

**Comment by mmusich** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**